libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -3701,6 +3701,76 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                     },
                     ReturnType.Number,
                     expr => ValidateArityAndAnyType(expr, 1, 1, ReturnType.String)),
+                new ExpressionEvaluator(
+                    ExpressionType.HasFullDate,
+                    (expr, state, options) =>
+                    {
+                        TimexProperty parsed = null;
+                        bool? value = null;
+                        string error = null;
+                        IReadOnlyList<object> args;
+                        (args, error) = EvaluateChildren(expr, state, options);
+                        if (error == null)
+                        {
+                            if (args[0] is TimexProperty timex)
+                            {
+                                parsed = timex;
+                            }
+                            else if (args[0] is JObject jTimex)
+                            {
+                                parsed = jTimex.ToObject<TimexProperty>();
+                            }
+                            else if (args[0] is string ts)
+                            {
+                                parsed = new TimexProperty(ts);
+                            } 
+                            else
+                            {
+                                error = $"{expr} should be either a TimexProperty or a string";
```

Comment:
>{expr} should be either a TimexProperty or a string [](start = 42, length = 51)

Not quite the right error string.  $"{expr} requires a TimexProperty or string as an argument."

Reply:
That's more accurate. Thanks